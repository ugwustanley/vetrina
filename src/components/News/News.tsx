import React from "react";
import { Box, Stack, Button } from "@mui/material";
import { Link } from "react-router-dom";
import { newsData } from "../../interface/index";
import Image from "./assets/image.svg";

type props = {
  data: newsData;
};

const News: React.FC<props> = ({ data }) => {
  const News = {
    minHeight: "80px",
    background: "#fff",
    position: "relative",
    display: "grid",
    gridTemplateColumns: "90px 1fr",
    boxSizing: "border-box",
  };
  const NewsContent = {
    // marginLeft:"1rem",
    boxSizing: "border-box",
    padding: ".9rem",
    paddingTop: "0",
    paddingBottom: "0",

    "& p": {
      color: "#103B66",
      fontSize: ".8rem",
      fontWeight: "normal",
      margin: 0,
      marginTop: ".2rem",
      height: "40px",
      textOverflow: "ellipsis",
      overflow: "hidden",
      width: "100%",

      //   whiteSpace: "nowrap",
    },
    "& h6": {
      fontSize: ".9rem",
      fontWeight: "normal",
      margin: 0,
      color: "#21B8F9",
      textTransform: "capitalize",
    },
    "& a": {
      fontSize: ".8rem",
      fontWeight: "lighter",
      margin: 0,
      color: "#103B66",
    },
  };

  const NewsImage = {
    backgroundImage: data?.image ? `url(${data?.image})` : `url(${Image})`,
    backgroundSize: "cover",
    //   backgroundColor:"red",
    width: "100%",
    height: "100%",
  };

  return (
    <Box sx={News} >
      <Box sx={NewsImage}></Box>
      <Box sx={NewsContent}>
        <h6 data-testid="news-item">{data?.category}</h6>
        <p>{data?.title}</p>
        <a href={`${data?.url}`}>Estimated reading: 4 min</a>
      </Box>
    </Box>
  );
};

export default News;
