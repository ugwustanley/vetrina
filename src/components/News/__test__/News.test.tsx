import React from 'react';
import { render, screen , cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import News from '../News';

afterEach(cleanup)

const data =  {
    author: "stanley",
    category: "General",
    country: "Nigeria",
    description: "This is a test description",
    image: "example.svg",
    language: "en",
    published_at: "Nigerian publisher",
    source: "cnn.com",
    title: "Example",
    url: "example.com/url",
  }


it('renders element without any error', () => {

  const div = document.createElement('div')

  ReactDOM.render(<News data={data} />, div)

});



// it('renders element without any error', () => {
   
//     render(<News data={data} />);
  
//     const linkElement = screen.getByText(/learn react/i);
  
//     expect(linkElement).toBeInTheDocument();
  
//   });


it("renders news item correctly", () =>{

    const {getByTestId} = render(<News data={data} />)

    expect(getByTestId("news-item")).toHaveTextContent("General")
})

//snapshot testing
it("matches snapshot", () =>{

    const tree = renderer.create(<News data={data} />).toJSON()

    expect(tree).toMatchSnapshot()
})





