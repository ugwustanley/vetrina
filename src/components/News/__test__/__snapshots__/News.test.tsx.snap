// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root css-1ql4si4"
>
  <div
    className="MuiBox-root css-13ff2pu"
  />
  <div
    className="MuiBox-root css-2kz2qg"
  >
    <h6
      data-testid="news-item"
    >
      General
    </h6>
    <p>
      Example
    </p>
    <a
      href="example.com/url"
    >
      Estimated reading: 4 min
    </a>
  </div>
</div>
`;
